test/cpp/api/misc.cpp

```diff
@@ -53,6 +53,21 @@ TEST(NNInitTest, CanInitializeTensorThatRequiresGrad) {
   ASSERT_EQ(torch::nn::init::ones_(tensor).sum().item<int32_t>(), 12);
 }
 
+TEST(NNInitTest, CalculateGainWithTanh) {
+  double gain = torch::nn::init::calculate_gain(torch::nn::init::Nonlinearity::Tanh);
+  ASSERT_EQ(gain, 5.0/3.0);
```

Comment:
Exact floating point equality tests are prone to flakiness due to numerical precision. You should use `ASSERT_DOUBLE_EQ`, see https://github.com/abseil/googletest/blob/master/googletest/docs/advanced.md#floating-point-macros